src/Adapter/OrderReturn/CommandHandler/UpdateOrderReturnStateHandler.php

```diff
@@ -0,0 +1,101 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\CommandHandler;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\Repository\OrderReturnRepository;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\Validator\OrderReturnValidator;
+use PrestaShop\PrestaShop\Adapter\OrderReturnState\Repository\OrderReturnStateRepository;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\CommandHandler\UpdateOrderReturnStateHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnException;
+
+class UpdateOrderReturnStateHandler implements UpdateOrderReturnStateHandlerInterface
+{
+    /**
+     * @var OrderReturnRepository
+     */
+    private $orderReturnRepository;
+
+    /**
+     * @var OrderReturnStateRepository
+     */
+    private $orderReturnStateRepository;
+
+    /**
+     * @var OrderReturnValidator
+     */
+    private $orderReturnValidator;
+
+    /**
+     * UpdateOrderReturnStateHandler constructor.
+     *
+     * @param OrderReturnRepository $orderReturnRepository
+     * @param OrderReturnStateRepository $orderReturnStateRepository
+     * @param OrderReturnValidator $orderReturnValidator
+     */
+    public function __construct(
+        OrderReturnRepository $orderReturnRepository,
+        OrderReturnStateRepository $orderReturnStateRepository,
+        OrderReturnValidator $orderReturnValidator
+    ) {
+        $this->orderReturnRepository = $orderReturnRepository;
+        $this->orderReturnStateRepository = $orderReturnStateRepository;
+        $this->orderReturnValidator = $orderReturnValidator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateOrderReturnStateCommand $command): void
+    {
+        $orderReturnId = $command->getOrderReturnId();
+        $orderReturn = $this->orderReturnRepository->get($orderReturnId);
+        $orderReturn = $this->updateOrderReturnWithCommandData($orderReturn, $command);
+
+        $this->orderReturnValidator->validate($orderReturn);
```

Comment:
You thumbed up, but it still needs to be done ^^ Or actually you handled it in the repository but still need to clean the code here no

Reply:
Yeah sorry I was thumbing up as I was adding changes, but I had not pushed commit yet. It was outdated because before it seems you did review at the same time i commited changes.